CODING.md

```diff
@@ -0,0 +1,83 @@
+# Juvix coding style guidelines
+
+## General rules
+
+- always use the formatter
+- don't use unicode, except perhaps for judoc
+- avoid `Pair`
+- prefer enumarations with meaningful names over booleans, e.g., instead of using `Bool` define:
+```
+	type Transferability := Transferable | NonTransferable
+```
+
+## Imports
+
+- all imports at the beginning of the file
+- explicit names for imported identifiers (`import X open using {iden1; iden2}`)
+    - exception: `import Stdlib.Prelude open`, `import Anoma open` are allowed
+
+## Names
+
+- types: upper camel case (`Nat`, `PublicKey`)
+- functions, constructors, etc: lower camel case (`tokenLogic`, `transferLogic`)
+- instances: lower came case with `I` at the end (`instance eqNatI : Eq Nat := ..`)
+- boolean check functions: start with `is` (`isWhatever`)
+- constructors: `mk` + type name (`mkResource`)
```

Comment:
I guess we should clarify this rule would only apply to record types with a single constructor